Content.Shared/Traits/Assorted/PermanentBlindnessSystem.cs

```diff
@@ -36,6 +37,17 @@ private void OnShutdown(Entity<PermanentBlindnessComponent> blindness, ref Compo
         _blinding.UpdateIsBlind(blindness.Owner);
     }
 
+    private void OnRemove(Entity<PermanentBlindnessComponent> blindness, ref ComponentRemove args)
+    {
+        if (!_entityManager.TryGetComponent<BlindableComponent>(blindness, out var blindable))
+            return;
+
+        if (blindable.MinDamage != 0)
+        {
+            _blinding.SetMinDamage(new Entity<BlindableComponent?>(blindness.Owner, blindable), 0);
+        }
+    }
```

Comment:
```suggestion
    private void OnShutdown(Entity<PermanentBlindnessComponent> blindness, ref ComponentShutdown args)
    {
        if (!TryComp<BlindableComponent>(blindness, out var blindable))
            return;

        if (blindable.MinDamage != 0)
        {
            _blinding.SetMinDamage((blindness.Owner, blindable), 0);
        }
    }
```
- We can simply replace `OnShutdown` with this I think. `ComponentRemove` is raised directly after `ComponentShutdown` in the next step of the component's lifestage, but we don't need both here. 
- `SetMinDamage` already includes `UpdateIsBlind` so we don't need that anymore
- Use the shorthand `TryComp` for better readability
- `(blindness.Owner, blindable)` does the same as using the full constructor, but is more readable

Could you make the last two adjustments for `OnMapInit` as well?

Reply:
updated!